juju/juju
# state: Database and modelBackend refactoring

## Description of change

In order to support new State-like types (coming soon) and allow them to leverage existing functionality in the state package, some low level txn functionality which was on State has been moved onto Database. Additionally the modelBackend interface has been extended to allow it to be used more widely, providing access to the underlying Database.

The state settings and watcher implementations are now completely based on modelBackend instead of State, allowing alternative State types to use these areas too.

## QA steps

This change shouldn't affect external behaviour. Unit tests still pass and a simple bootstrap, deploy, destroy check was undertaken to ensure basic functionality is still OK.

## Documentation changes

Not required - no used visible changes.

## Bug reference

N.A.
